### ISSUE TITLE
Drill Time Range Filter not usable

### ISSUE THREAD

**Issue by Z0ltrix** [apache/superset#18869]
we have a problem with our superset -> drill connection with time range filters.

When we filter a dashboard by time range (last week, month, etc.) i get an 

```
SYSTEM ERROR: ClassCastException: org.apache.drill.exec.expr.holders.NullableTimeStampHolder cannot be cast to org.apache.drill.exec.expr.holders.TimeStampHolder
```

from drill.

#### How to reproduce the bug

1. Create a Dataset based on Drill
2. Create a Filter Box
3. Create a Dashboard
4. Make sure everything works 
5. Aktivate a Time Range Filter on the Filter Box
6. Everything fails with the error in Drill

### Expected results

Superset should send the following filter:
```
WHERE `startTime` >= TIMESTAMP '2022-02-14 00:00:00.000000'
  AND `startTime` < TIMESTAMP '2022-02-21 00:00:00.000000'
ORDER BY `startTime` DESC
```
### Actual results

Superset sends the following to drill:
```
WHERE `startTime` >= '2022-02-14 00:00:00.000000'
  AND `startTime` < '2022-02-21 00:00:00.000000'
ORDER BY `startTime` DESC
```

#### Screenshots

![image](https://user-images.githubusercontent.com/14264280/155306427-8cd7c818-1009-41b7-92fc-c087ee9d0c9d.png)
![image](https://user-images.githubusercontent.com/14264280/155306488-1ca61a5f-95c6-45e8-9bbf-288d8fc92ae8.png)

### Environment

(please complete the following information):

- browser type and version: Chrome Version 98.0.4758.102
- superset version: 1.3.0
- python version: 3.8.0

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [x] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [ ] I have reproduced the issue with at least the latest released version of superset.
- [ ] I have checked the issue tracker for the same issue and I haven't found one similar.
- [x] I have discussed the question on dev mailing list

**Comment by villebro** [comment]
Thanks for reporting @Z0ltrix - can you check what the datatype of `startDate` is in your dataset (=the native column type in the physical table)?

**Comment by Z0ltrix** [comment]
`select typeof(startTime) from dfs.path.dataset limit 1` returns TIMESTAMP 

in Superset this is shown as DATETIME
![image](https://user-images.githubusercontent.com/14264280/155315552-b233b821-217f-4087-a67b-52a6b23acd93.png)

**Comment by jnturton** [comment]
Hi, both of the syntaxes exhibited by @Z0ltrix above should work with a TIMESTAMP column but due to a bug affecting an optional Drill feature (which @Z0ltrix had enabled) the error shown is encountered in one case.  We're going to rework the Drill feature so that this problem goes away and Superset's SQL syntax will not cause any trouble.

https://issues.apache.org/jira/browse/DRILL-8157

The immediate workaround for anyone affected by this is ``SET `functions.cast_empty_string_to_null` = false;``

**Comment by rusackas** [comment]
Since this thread has an inbould solution as of a year ago, an apparent workaround, and seemingly no additional complaints in upward of a year, I'm going to close this one out. We also no longer intend to support the FilreBox going forward, in favor of Dashboard Native FIlters. If anyone has similar issues with those updated filters in modern (1.5.x, 2.0.x) versions of Superset, say the word, and we can open a new ticket with updated context.

**Comment by jnturton** [comment]
@rusackas @Z0ltrix and future readers: according to my testing this issue is resolved as of Drill 1.21.
